packages/@glimmer-workspace/integration-node-tests/test/prettier.test.ts

```diff
@@ -0,0 +1,207 @@
+import { mkdtemp, readFile, writeFile } from 'node:fs/promises';
+import { createRequire } from 'node:module';
+import os from 'node:os';
+import { join } from 'node:path';
+
+import { $ } from 'execa';
+import * as prettier from 'prettier';
+import { describe, expect, it } from 'vitest';
+
+const monorepoRoot = join(import.meta.dirname, '../../../../');
+const require = createRequire(import.meta.url);
+
+/**
+ * See: https://github.com/glimmerjs/glimmer-vm/issues/1688
+ *
+ * Requires the root package.json#pnpm#overrides point at our internal
+ * copy of @glimmer/syntax, or else prettier brings its own already published
+ * copy of @glimmer/syntax
+ */
+describe('Prettier', () => {
+  it(`SANITY: we've symlinked to the in-repo copy of @glimmer/syntax`, () => {
+    let workspacePath = require.resolve('@glimmer/syntax');
+    let prettierPath = require.resolve('prettier');
+    let prettierGlimmer = require.resolve('@glimmer/syntax', { paths: [prettierPath] });
+
+    expect(prettierGlimmer).toBe(workspacePath);
+  });
+
+  it('Underlynig preprocess API works', async () => {
+    let result = (await import('@glimmer/syntax')).preprocess('<h1></h1>');
+
+    expect(result, `It can be await import()'d, and doesn't error`).toBeTruthy();
+  });
+
+  it('Prettier can call preprocess', async () => {
+    let result = await prettier.format(`     <div>\n</div>`, { parser: 'glimmer' });
+
+    expect(result).toMatchInlineSnapshot(`
+      "<div>
+      </div>"
+    `);
+  });
+
+  /**
+   * This is important because we don't test the code we ship to npm
+   * (the code we ship to npm goes through a build process, and we opted
+   *  to not do that for in-repo dev ergonomics)
+   *
+   *  Process:
+   *  1. build @glimmer/syntax (and dependencies)
+   *  2. creat a tmp folder
+   *  3. create a project in that tmp folder
+   *  4. pack and add the tarballs to the project
+   *  5. See if a basic import and call to pre-process works
+   */
+  it('Uses the real build, and not our monorepo infra', { timeout: 20_000 }, async () => {
+    /**
+     * pnpm packs tgz's with the name format ${hyphenated-package-name}-${version}.tgz
+     */
+    async function versionOf(name) {
+      let manifestPath = join(monorepoRoot, 'packages', name, 'package.json');
+      let manifestString = await readFile(manifestPath);
+      return JSON.parse(manifestString.toString()).version;
+    }
+
+    let tarballs = {
+      syntax: `glimmer-syntax-${await versionOf('@glimmer/syntax')}`,
+      util: `glimmer-util-${await versionOf('@glimmer/util')}`,
+      wireFormat: `glimmer-wire-format-${await versionOf('@glimmer/wire-format')}`,
+    };
+
+    let file = `console.log((await import('@glimmer/syntax')).preprocess('<h1></h1>'));`;
+    let manifest = JSON.stringify({
+      name: 'real-test',
+      type: 'module',
+      private: true,
+      devDependencies: {
+        '@glimmer/syntax': `file://./${tarballs.syntax}`,
```

Comment:
Should we be consolidating this with the equivalent work I did for the benchmark? It would be awesome to have reusable infra for making tests with the published packages, and we need it already for benchmarks. (not a blocker, and I could do the follow-up work if you think it's a good idea).

Reply:
I like this plan, would clean up my test a bit. will update